src/data/s3-data-source.ts

```diff
@@ -64,9 +66,29 @@ export class S3DataSource implements ContiguousDataSource {
     });
 
     try {
+      // TODO: Use S3 client instead of axios when aws-lite supports Metadata on head-requests
+      // const head = this.s3Client.HeadObject({
+      //   Bucket: this.s3Bucket,
+      //   Key: `${this.s3Prefix}/${id}`,
+      // });
+      const head = await axios.head(
+        `${AWS_ENDPOINT}/${this.s3Bucket}/${this.s3Prefix}/${id}`,
+        { validateStatus: () => true },
```

Comment:
It should return a `200` when success, right? If not can we have a more specific here?

Reply:
this allows us to provide a better error message on cache misses. where we say the s3 head check has failed rather than  `AxiosError`

Reply:
fwiw, this class could log it as `warn` or `debug` at this level since it gets logged above